packages/tables/src/Columns/Concerns/CanBeSelectable.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Filament\Tables\Columns\Concerns;
+
+use Illuminate\Support\Str;
+
+trait CanBeSelectable
+{
+    protected bool $isSelectable = false;
+
+    public function selectable(bool | array $condition = true): static
+    {
+        if (is_array($condition)) {
+            $this->isSelectable = true;
+        } else {
+            $this->isSelectable = $condition;
+        }
+
+        return $this;
+    }
+
+    public function isSelectable(): bool
+    {
+        return $this->isSelectable;
```

Comment:
If we add support for closures ^ then we need to send this through `$this->evaluate()`